turbo/snapshotsync/freezeblocks/block_reader.go

```diff
@@ -458,17 +500,13 @@ func (r *BlockReader) HeaderByHash(ctx context.Context, tx kv.Getter, hash commo
 var emptyHash = common.Hash{}
 
 func (r *BlockReader) CanonicalHash(ctx context.Context, tx kv.Getter, blockHeight uint64) (h common.Hash, err error) {
-	h, err = rawdb.ReadCanonicalHash(tx, blockHeight)
-	if err != nil {
-		return h, err
+	// Handle nil case with db hit
+	if r == nil {
+		return rawdb.ReadCanonicalHash(tx, blockHeight) // Fallback to db if not found in snapshots
 	}
-	if h != emptyHash {
-		return h, nil
-	}
-
 	seg, ok, release := r.sn.ViewSingleFile(coresnaptype.Headers, blockHeight)
 	if !ok {
-		return
+		return rawdb.ReadCanonicalHash(tx, blockHeight) // Fallback to db if not found in snapshots
```

Comment:
for performance must be opposite: 
- check db first
- if not in db read from file (it's more expensive)

Reply:
done